src/vs/workbench/contrib/policyExport/electron-browser/policyExport.contribution.ts

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IWorkbenchContribution, registerWorkbenchContribution2, WorkbenchPhase } from '../../../common/contributions.js';
+import { Disposable } from '../../../../base/common/lifecycle.js';
+import { IWorkbenchConfigurationService } from '../../../services/configuration/common/configuration.js';
+import { IExtensionService } from '../../../services/extensions/common/extensions.js';
+import { INativeEnvironmentService } from '../../../../platform/environment/common/environment.js';
+import { ILogService } from '../../../../platform/log/common/log.js';
+import { INativeHostService } from '../../../../platform/native/common/native.js';
+import { Registry } from '../../../../platform/registry/common/platform.js';
+import { Extensions, IConfigurationRegistry } from '../../../../platform/configuration/common/configurationRegistry.js';
+import { IProgressService, ProgressLocation } from '../../../../platform/progress/common/progress.js';
+import { IFileService } from '../../../../platform/files/common/files.js';
+import { URI } from '../../../../base/common/uri.js';
+import { VSBuffer } from '../../../../base/common/buffer.js';
+import { PolicyCategory, PolicyCategoryData } from '../../../../base/common/policy.js';
+import { ExportedPolicyDataDto } from '../../../../base/common/policyDto.js';
+import { join } from '../../../../base/common/path.js';
+
+export class PolicyExportContribution extends Disposable implements IWorkbenchContribution {
+	static readonly ID = 'workbench.contrib.policyExport';
+	static readonly DEFAULT_POLICY_EXPORT_PATH = 'build/lib/policies/policyData.jsonc';
+
+	constructor(
+		@INativeEnvironmentService private readonly nativeEnvironmentService: INativeEnvironmentService,
+		@IExtensionService private readonly extensionService: IExtensionService,
+		@IFileService private readonly fileService: IFileService,
+		@IWorkbenchConfigurationService private readonly configurationService: IWorkbenchConfigurationService,
+		@INativeHostService private readonly nativeHostService: INativeHostService,
+		@IProgressService private readonly progressService: IProgressService,
+		@ILogService private readonly logService: ILogService,
+	) {
+		super();
+
+		// Skip for non-development flows
+		if (this.nativeEnvironmentService.isBuilt) {
+			return;
+		}
+
+		const policyDataPath = this.nativeEnvironmentService.exportPolicyData;
+		if (policyDataPath !== undefined) {
+			const defaultPath = join(this.nativeEnvironmentService.appRoot, PolicyExportContribution.DEFAULT_POLICY_EXPORT_PATH);
+			void this.exportPolicyDataAndQuit(policyDataPath ? policyDataPath : defaultPath);
```

Comment:
`void `? I'm guessing it's because of that function being called is `async`? Does the compile barf without this? Or was it just AI being overly cautious?

Reply:
this was mostly just me doing it out of habit actually, since my previous project had a lint rule for this 😆  I do think it helps visually see that we're calling async without waiting for it though

Reply:
I mean... I don't **_not_** like it. 😄